### PR TITLE
Add cursed item support to artifacts.xml

### PR DESCRIPTION
**Pull request by mkristofik**
And a couple extras:

- include artifacts.xml in the installer
- removed a weird variable that pointed to the artifact name list, off by one index

### CODE REVIEW

File: src/cpp/shared/artifacts.cpp [hunk 1]
```diff
@@ -70,6 +70,7 @@ unsigned char gArtifactLevel[NUM_SUPPORTED_ARTIFACTS] = { 0 };
 std::vector<std::string> names;
 std::vector<std::string> descriptions;
 std::vector<std::string> events;
+std::vector<char> isCursed;
```
Comment (jkoppel) [inline src/cpp/shared/artifacts.cpp]: Why is this a char?

Comment (jkoppel) [reply src/cpp/shared/artifacts.cpp]: I mean....why is it not vector<boolean> ?

Comment (mkristofik) [reply src/cpp/shared/artifacts.cpp]: Sorry, it looks like github munged what I wrote.  I meant that `vector<bool>` is commonly avoided and `vector<char>` is a useful alternative.  I'm using `char` as a 1-byte integer in this case. 
 `vector<int>` would work too but takes up more space.

Comment (jkoppel) [reply src/cpp/shared/artifacts.cpp]: http://www.codingstandard.com/rule/17-1-1-do-not-use-stdvector/

I see. Add a comment about that, and then merge this in yourself.

Comment (mkristofik) [reply src/cpp/shared/artifacts.cpp]: My next change already has changes to this file so I'll add it there.

**Pull request merged**